domain/access/modelmigration/export_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmigration
+
+import (
+	"context"
+	"time"
+
+	"github.com/juju/description/v8"
+	"github.com/juju/names/v5"
+	jc "github.com/juju/testing/checkers"
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	coremodel "github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/juju/core/user"
+)
+
+type exportSuite struct {
+	coordinator *MockCoordinator
+	service     *MockExportService
+}
+
+var _ = gc.Suite(&exportSuite{})
+
+func (s *exportSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+
+	s.coordinator = NewMockCoordinator(ctrl)
+	s.service = NewMockExportService(ctrl)
+
+	return ctrl
+}
+
+func (s *exportSuite) newExportOperation() *exportOperation {
+	return &exportOperation{
+		service: s.service,
+	}
+}
+
+func (s *exportSuite) TestExport(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	dst := description.NewModel(description.ModelArgs{})
+
+	bobTag := names.NewUserTag("bob")
+	bobName := user.NameFromTag(bobTag)
+	bazzaTag := names.NewUserTag("bazza")
+	bazzaName := user.NameFromTag(bazzaTag)
+	steveTag := names.NewUserTag("steve")
+
+	userAccesses := []permission.UserAccess{{
+		UserTag:     bazzaTag,
+		Access:      permission.ReadAccess,
+		CreatedBy:   bobTag,
+		DateCreated: time.Now(),
+		DisplayName: bazzaName.Name(),
+		UserName:    bazzaName,
+	}, {
+		UserTag:     bobTag,
+		Access:      permission.AdminAccess,
+		CreatedBy:   steveTag,
+		DateCreated: time.Now(),
+		DisplayName: bobName.Name(),
+		UserName:    bobName,
+	}}
+
+	s.service.EXPECT().ReadAllUserAccessForTarget(gomock.Any(), permission.ID{
+		ObjectType: permission.Model,
+		Key:        dst.Tag().Id(),
+	}).Return(userAccesses, nil)
+
+	bobTime := time.Now().Round(time.Minute).UTC()
```

Comment:
**todo:** Add a comment to explain the use case. It is painful to review it (or debug it) when we have to figure out ourselves that bob as logged a minute before bazza, and try to infer the intent (maybe it is important, maybe we just want different login time, i don't know ;) ) 

Usually explicit intent is a good thing, especially in UT (when it break, we don't want to take time to figure out what it is testing.)

Reply:
There's no reason for these to be one minute after the other, I've changed the code so that they at least don't depend on each other, that hopefully makes it clearer.